### DISCUSSION TITLE
Real runtime config

### DISCUSSION THREAD

**Discussion by tbilek** [vercel/next.js#20424] [Help]
Hi,

I have a problem with real runtime configuration and I am not able to find a solution.

This is my target :)
I would like to have a one build which I am able to deploy to any environemnts without changing the build itself.

Ussualy I would solve this problem by `window` object where I would set the property (for example API endpoint) which I would use in code, but in Next.js, it does not work of course...

I use `next build && next export` for creating the build.

Thanks,
Tomas

**Comment by rafaelalmeidatk** [comment]
`next export` will generate a static build that cannot be changed later. You can however inject properties in the `window` object as you mentioned. Since you are reading these values at runtime you cannot statically build the components that need these properties to render, they would need to render some placeholder until they run in the browser to have access to the `window`
